### DISCUSSION TITLE
Fake path of csproj

### DISCUSSION THREAD

**Discussion by timcassell** [dotnet/msbuild#9056] [Q&A]
Is it possible to override the path msbuild uses when parsing a csproj? I want to pretend like the csproj is at a different directory, so that relative paths will be resolved at that other directory. I would move it before running msbuild, but I might not have permissions to do so.

Context: In BenchmarkDotNet, we are looking at copying the user's original csproj to resolve a host of issues in the auto generated project. If we just copy it and build it as-is, all relative paths will be broken. And there is no easy way for us to just parse it and rewrite the relative paths ourselves that I know of, especially if they include imports. https://github.com/dotnet/BenchmarkDotNet/issues/1403

**Comment by danmoseley** [comment]
Not that I am aware of.

Why would you not have permissions to copy it, if you have permissions to build it (read it) apparently?

**Reply by timcassell** [reply]
We can have read permissions without write permissions. BDN provides an API to specify an output path when the default path is inaccessible. https://github.com/dotnet/BenchmarkDotNet/issues/377 I want to be able to write the generated csproj to that path, but resolve relative paths at the original path.

Also, it would just be cleaner if we didn't write generated files to the user's project directory.
